cmd/switch.go

```diff
@@ -36,34 +37,85 @@ func newSwitchCmd(homedir string) *cobra.Command {
 		Example: `# switch the active istioctl version to version=1.7.4, flavor=tetrate and flavor-version=1
 $ getistio switch --version 1.7.4 --flavor tetrate --flavor-version=1`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			d := &api.IstioDistribution{
-				Version:       flagVersion,
-				Flavor:        flagFlavor,
-				FlavorVersion: int64(flagFlavorVersion),
+			d, err := switchParse(homedir, flagName, flagVersion, flagFlavor, flagFlavorVersion)
+			if err != nil {
+				return err
 			}
-
 			return switchExec(homedir, d)
 		},
 	}
 
 	flags := cmd.Flags()
 	flags.SortFlags = false
-	flags.StringVarP(&flagVersion, "version", "", "", "Version of istioctl e.g. 1.7.4")
+	flags.StringVarP(&flagName, "name", "", "", "Name of istioctl, e.g. 1.9.0-istio-v0")
+	flags.StringVarP(&flagVersion, "version", "", "", "Version of istioctl, e.g. 1.7.4")
 	flags.StringVarP(&flagFlavor, "flavor", "", "", "Flavor of istioctl, e.g. \"tetrate\" or \"tetratefips\" or \"istio\"")
 	flags.IntVarP(&flagFlavorVersion, "flavor-version", "", -1, "Version of the flavor, e.g. 1")
 
-	_ = cmd.MarkFlagRequired("version")
-	_ = cmd.MarkFlagRequired("flavor")
-	_ = cmd.MarkFlagRequired("flavor-version")
-
 	return cmd
 }
 
+// if set name, it should only parse name to distro
+// if version, flavor and version are all set, just parse it to distro
+// if there exists active distro, switch with only one or two command will use the active distro setting for unset command
+// if there are no active distro exists, switch with only one or two command will use the default distro setting for unset command
+// if all commands are not set, use active setting if there has otherwise use default version
+// default version: latest version, default flavor: tetrate, default flavorversion: 0
+func switchParse(homedir, flagName, flagVersion, flagFlavor string, flagFlavorVersion int) (*api.IstioDistribution, error) {
+	if len(flagName) != 0 {
+		d, err := api.IstioDistributionFromString(flagName)
+		if err != nil {
+			logger.Infof("cannot parse given name to %s istio distribution\n", flagName)
+			return nil, err
+		}
+		return d, err
+	}
+	fetched, err := istioctl.GetFetchedVersions(homedir)
+	if err != nil {
+		logger.Infof("cannot fetch istio manifest\n")
+		return nil, err
+	}
+
+	currDistro, err := istioctl.GetCurrentExecutable(homedir)
+	if err != nil {
+		return switchHandleDistro(nil, fetched[0].Version, flagVersion, flagFlavor, flagFlavorVersion)
+	}
+	return switchHandleDistro(currDistro, fetched[0].Version, flagVersion, flagFlavor, flagFlavorVersion)
```

Comment:
I think we can simplify:
```suggestion
	currDistro, _ := istioctl.GetCurrentExecutable(homedir)
	return switchHandleDistro(currDistro, fetched[0].Version, flagVersion, flagFlavor, flagFlavorVersion)
```